hub/integrations/wandb/wandb.py

```diff
@@ -0,0 +1,250 @@
+        if key in _CREATED_DATASETS:
+            artifact = artifact_from_ds(ds)
+            wandb_info = ds.info.get("wandb") or {"commits": {}}
+            commits = wandb_info["commits"]
+            info = {}
+            commits[ds.commit_id] = info
+            info["created-by"] = {
+                "run": {
+                    "entity": run.entity,
+                    "project": run.project,
+                    "id": run.id,
+                    "url": run.url,
+                },
+                "artifact": artifact.name,
+            }
+            ds.info["wandb"] = wandb_info
+            ds.flush()
+            _CREATED_DATASETS.remove(key)
+            run.log_artifact(artifact)
+    else:
+        _CREATED_DATASETS.discard(key)
+
+
+def dataset_read(ds):
+    path = ds.path
+    run = wandb_run()
+    if not run:
+        return
+    if run.id not in _READ_DATASETS:
+        _READ_DATASETS.clear()
+        _READ_DATASETS[run.id] = {}
+    keys = _READ_DATASETS[run.id]
+    key = get_ds_key(ds)
+    if key not in keys:
+        keys[key] = None
+        try:
+            input_datasets = run.config.input_datasets
+        except (KeyError, AttributeError):
+            input_datasets = []
+        dsconfig = dataset_config(ds)
+        path = dsconfig["Dataset"]
+        if dsconfig not in input_datasets:
+
+            if path.startswith("hub://"):
+                import wandb
+
+                run.log(
+                    {
+                        f"Hub Dataset [{path[len('hub://'):]}]": wandb.Html(
+                            viz_html(path), False
+                        )
+                    }
+                )
+
+            input_datasets.append(dsconfig)
+            run.config.input_datasets = input_datasets
+
+        # TODO consider optimized datasets:
+        wandb_info = ds.info.get("wandb", {}).get("commits", {}).get(ds.commit_id)
+        if wandb_info:
+            run_and_artifact = wandb_info["created-by"]
+            run_info = run_and_artifact["run"]
+            artifact = run_and_artifact["artifact"]
+            run.use_artifact(
+                f"{run_info['entity']}/{run_info['project']}/{artifact}:latest"
+            )
+        else:
+            # For datasets that were not created during a wandb run,
+            # we want to "use" an artifact that is not logged by any run.
+            # This is not possible with wandb yet.
+
+            # artifact = artifact_from_ds(ds)
+            # run.use_artifact(artifact)
+            pass
+
+
+def viz_html(hub_path: str):
+    #     return f"""
+    #       <div id='container'></div>
+    #   <script src="https://app.activeloop.ai/visualizer/vis.js"></script>
+    #   <script>
+    #     let container = document.getElementById('container')
+
+    #     window.vis.visualize('{hub_path}', null, null, container, {{
+    #       requireSignin: true
+    #     }})
+    #   </script>
+    #     """
```

Comment:
What this commented lines of core represent? If it's the part of the code that is no longer used and not going to be, then we should remove these lines of code